Add CREF about ambiguous behaviour of additionalProperties

Relequestual opened the pull request:
Fixes #1172
Must see new issue relating to the behaviour of annotation collection for resolving in the next draft.
Related to https://github.com/json-schema-org/community/discussions/57

karenetheridge commented on jsonschema-core.xml:
```diff
@@ -2480,6 +2480,28 @@
                             Implementations that do not support annotation collection MUST do so.
+                            <cref>
+                                In defining this option, it seems there is the potential for
+                                ambiguity in the output format. The ambiguity does not effect validation results,
+                                but it does effect the resutling output format.
```
```suggestion
                                ambiguity in the output format. The ambiguity does not affect validation results,
                                but it does affect the resulting output format.
```

karenetheridge commented on jsonschema-core.xml:
```diff
@@ -2480,6 +2480,28 @@
+                                https://github.com/json-schema-org/community/discussions/57 for details.
+                                We discussed the possible solutions at length (https://github.com/json-schema-org/json-schema-spec/issues/1172).
+                                We concluded that we need to re-evaluate the annotation collection system
+                                and its use for validation purposes, but we also agreed not for this release.
```
```suggestion
                                and its use for validation purposes, but we also agreed not to do so for this release.
```

karenetheridge commented on jsonschema-core.xml:
```diff
@@ -2480,6 +2480,28 @@
+                                In defining this option, it seems there is the potential for
+                                ambiguity in the output format. The ambiguity does not effect validation results,
+                                but it does effect the resutling output format.
+                                The ambiguity allows for multiple valid output results depending on if annotations
```
```suggestion
                                The ambiguity allows for multiple valid output results depending on whether annotations
```

karenetheridge commented on jsonschema-core.xml:
```diff
@@ -2480,6 +2480,28 @@
+                                and its use for validation purposes, but we also agreed not for this release.
+                                This confusion stemmed from the understanding of "evaluated" and the impact on
+                                annotation collection.
+                                It was proposed that we could read a different interpritation (https://github.com/json-schema-org/json-schema-spec/issues/1172#issuecomment-1049686478)
+                                however the previous interpritation is still valid.
```
```suggestion
                                It was proposed that we could read a different interpretation (https://github.com/json-schema-org/json-schema-spec/issues/1172#issuecomment-1049686478)
                                however the previous interpretation is still valid.
```

Relequestual commented:
I've removed some of the waffle from the CREF in the spec document as @jdesrosiers requested.
I've also added an Architectural Decision Record (ADR) for this overall decision to capture the reasoning behind it.
As you are named on the ADR (@karenetheridge @jdesrosiers @gregsdennis), it would be great to get your reviews.

Please feel free to make suggestion changes in your reviews if any additional thoughs come to mind or I've forgotten an option we considered.

karenetheridge reviewed (approved):
+1. Moving the decision content to an ADR is a good move as it allows more verbosity as well as linking to all the related issues where the discussions took place.

jdesrosiers commented on adr/2022-04-08-cref-for-ambiguity-and-fix-later-gh-spec-issue-1172.md:
```diff
@@ -0,0 +1,87 @@
+We needed to decide on how to proceed for the patch release of the 2020-12 version of the specification.
+
+The two above links are to a GitHub Discussion and a GitHub Issue detailing the problems.
+Details with an example of the problem can be see in the Discussion's opening post specifically.
```
```suggestion
Details with an example of the problem can be seen in the Discussion's opening post specifically.
```

jdesrosiers reviewed (approved):
Looks good. Creating an ADR for this was a great idea! If we had one for when the annotation based behavior was was introduced, we could have saved ourselves a lot of speculation about what the intention was.

gregsdennis commented on adr/2022-04-08-cref-for-ambiguity-and-fix-later-gh-spec-issue-1172.md:
```diff
@@ -0,0 +1,87 @@
+## Considered Options
+
+* [Leaving it "as is" and do nothing](https://github.com/json-schema-org/community/discussions/57#discussioncomment-1413777)
+* [Pick one]()https://github.com/json-schema-org/community/discussions/57#discussioncomment-1416683) of the behaviours
```
Link formatting

gregsdennis reviewed (approved):
Minor formatting issue. Good otherwise.

Relequestual commented:
Rebased to remove commits for spelling and grammar fixes.
